ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ProposersDataManager.java

```diff
@@ -199,11 +199,17 @@ public SafeFuture<Optional<PayloadBuildingAttributes>> calculatePayloadBuildingA
     return getStateInEpoch(epoch)
         .thenApplyAsync(
             maybeState ->
-                calculatePayloadBuildingAttributes(blockSlot, epoch, maybeState, mandatory),
+                calculatePayloadBuildingAttributes(
+                    forkChoiceUpdateData.getForkChoiceState().getHeadBlockRoot(),
```

Comment:
I would just initialise it before Bytes32 parentBeaconBlockRoot = ... to bring more clarity.